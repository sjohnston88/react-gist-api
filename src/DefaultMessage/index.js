import React, {Component} from 'react';
class DefaultMessage extends Component {
  
  render(){
    return (
      <div> 
        <h3>Welcome to React-Gists-API App</h3>
        <p>By default you'll see GitHub user defunkts gists. You can open the drawer on the left to view more Gists.</p>
        <p>Enter a username in the URL bar in place of defunkts to fetch different Gists. Additionally, you can provide a <code>?since=YYYY-MM-DD</code> parameter to filter results by date.</p>
      </div>
    );
  }
}

export default DefaultMessage;