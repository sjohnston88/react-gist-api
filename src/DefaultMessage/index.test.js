import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import DefaultMessage from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Component/DefaultMessage', () => {
  const setup = () => shallow(<DefaultMessage />);

  it('renders correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot('default');
  });

});
