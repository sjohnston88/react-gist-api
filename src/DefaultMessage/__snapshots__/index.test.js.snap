// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DefaultMessage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Welcome to React-Gists-API App
</h3>,
        <p>
          By default you'll see GitHub user "defunkts" Gists. Open the left drawer to view more Gists from this user.
</p>,
        <p>
          Enter a username in the URL bar in place of "defunkts" to fetch different Gists. Additionally, you can provide a 
          <code>
                    ?since=YYYY-MM-DD
          </code>
           parameter to filter results by date.
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to React-Gists-API App",
        },
        "ref": null,
        "rendered": "Welcome to React-Gists-API App",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "By default you'll see GitHub user \\"defunkts\\" Gists. Open the left drawer to view more Gists from this user.",
        },
        "ref": null,
        "rendered": "By default you'll see GitHub user \\"defunkts\\" Gists. Open the left drawer to view more Gists from this user.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Enter a username in the URL bar in place of \\"defunkts\\" to fetch different Gists. Additionally, you can provide a ",
            <code>
              ?since=YYYY-MM-DD
</code>,
            " parameter to filter results by date.",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Enter a username in the URL bar in place of \\"defunkts\\" to fetch different Gists. Additionally, you can provide a ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "?since=YYYY-MM-DD",
            },
            "ref": null,
            "rendered": "?since=YYYY-MM-DD",
            "type": "code",
          },
          " parameter to filter results by date.",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Welcome to React-Gists-API App
</h3>,
          <p>
            By default you'll see GitHub user "defunkts" Gists. Open the left drawer to view more Gists from this user.
</p>,
          <p>
            Enter a username in the URL bar in place of "defunkts" to fetch different Gists. Additionally, you can provide a 
            <code>
                        ?since=YYYY-MM-DD
            </code>
             parameter to filter results by date.
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to React-Gists-API App",
          },
          "ref": null,
          "rendered": "Welcome to React-Gists-API App",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "By default you'll see GitHub user \\"defunkts\\" Gists. Open the left drawer to view more Gists from this user.",
          },
          "ref": null,
          "rendered": "By default you'll see GitHub user \\"defunkts\\" Gists. Open the left drawer to view more Gists from this user.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Enter a username in the URL bar in place of \\"defunkts\\" to fetch different Gists. Additionally, you can provide a ",
              <code>
                ?since=YYYY-MM-DD
</code>,
              " parameter to filter results by date.",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Enter a username in the URL bar in place of \\"defunkts\\" to fetch different Gists. Additionally, you can provide a ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "?since=YYYY-MM-DD",
              },
              "ref": null,
              "rendered": "?since=YYYY-MM-DD",
              "type": "code",
            },
            " parameter to filter results by date.",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
